pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -660,19 +661,19 @@ private void doAuthentication(PGStream pgStream, String host, String user, Prope
                   LOGGER.log(Level.FINEST, " <=BE AuthenticationReqMD5(salt={0})", Utils.toHexString(md5Salt));
                 }
 
-                byte[] encodedPassword = AuthenticationPluginManager.getEncodedPassword(AuthenticationRequestType.MD5_PASSWORD, info);
-                byte[] digest =
-                    MD5Digest.encode(user.getBytes(StandardCharsets.UTF_8), encodedPassword, md5Salt);
+                AuthenticationPluginManager.withEncodedPassword(AuthenticationRequestType.MD5_PASSWORD, info, encodedPassword -> {
+                  byte[] digest = MD5Digest.encode(user.getBytes(StandardCharsets.UTF_8), encodedPassword, md5Salt);
```

Comment:
Should withEncodedPassword be confined to `MD5Digest.encode` only?

Reply:
The cleartext code uses it as well.

Reply:
Can you please clarify?
You do not need encodedPassword past MD5Digest.encode, so why make a bigger withEncodedPassword block?

Reply:
Oh I thought you were asking why that method exists and meant that it's also used when sending the plaintext password.

In all those cases I wrapped the entire sub-block that is using the password or encoded password to keep the changeset simpler.

Reply:
The intention should be to minimize the use of the password, and erase it as soon as it is no longer needed.

In this case, making the block smaller yields easier to follow changeset.

However, I guess we need to erase md5-encoded block as well, as it basically allows relay attack.